samza-test/src/main/java/org/apache/samza/test/framework/system/CollectionStreamSystemSpec.java

```diff
@@ -51,11 +55,13 @@
    * <p>
    * @param systemName represents unique name of the system
    */
-  private CollectionStreamSystemSpec(String systemName) {
+  private CollectionStreamSystemSpec(String systemName, String jobName) {
     this.systemName = systemName;
     systemConfigs = new HashMap<String, String>();
     systemConfigs.put(String.format(SYSTEM_FACTORY, systemName), InMemorySystemFactory.class.getName());
```

Comment:
Is this required, why do we still need this if we have an override ?

Reply:
Override only applies at the end of planning. System Factory is still required during planning to get stream metadata. So we need to override both.